packages/protocol/test-sol/common/FeeHandler.t.sol

```diff
@@ -0,0 +1,906 @@
+
+    feeHandler.activateToken(address(stableToken));
+    assertTrue(feeHandler.getTokenActive(address(stableToken)));
+    address[] memory expectedActiveTokens = new address[](1);
+    expectedActiveTokens[0] = address(stableToken);
+    assertEq(feeHandler.getActiveTokens(), expectedActiveTokens);
+  }
+}
+
+contract FeeHandlerSetFeeBeneficiary is FeeHandlerFoundry {
+  function test_OnlyOwnerCanSetFeeBeneficiary() public {
+    vm.prank(user);
+    vm.expectRevert("Ownable: caller is not the owner");
+    feeHandler.setFeeBeneficiary(EXAMPLE_BENEFICIARY_ADDRESS);
+  }
+
+  function test_SetsAddressCorrectly() public {
+    feeHandler.setFeeBeneficiary(EXAMPLE_BENEFICIARY_ADDRESS);
+    assertEq(feeHandler.feeBeneficiary(), EXAMPLE_BENEFICIARY_ADDRESS);
+  }
+}
+
+contract FeeHandlerDistribute is FeeHandlerFoundry {
+  modifier setUpBeneficiary() {
+    feeHandler.setFeeBeneficiary(EXAMPLE_BENEFICIARY_ADDRESS);
+    _;
+  }
+
+  modifier activateToken() {
+    feeHandler.addToken(address(stableToken), address(mentoSeller));
+    feeHandler.activateToken(address(stableToken));
+    _;
+  }
+
+  modifier fundFeeHandlerStable(uint256 stableAmount) {
+    uint256 celoAmount = 1e18;
+    celoToken.approve(address(exchangeUSD), celoAmount);
+    exchangeUSD.sell(celoAmount, 0, true);
+    stableToken.transfer(address(feeHandler), stableAmount);
+    _;
+  }
+
+  modifier setBurnFraction() {
+    feeHandler.setBurnFraction(FixidityLib.newFixedFraction(80, 100).unwrap());
+    _;
+  }
+
+  modifier setMaxSlippage() {
+    feeHandler.setMaxSplippage(address(stableToken), FIXED1);
+    _;
+  }
+
+  function test_CantDistributeWhenNotActive() public setUpBeneficiary {
+    vm.expectRevert("Handler has to be set to sell token");
+    feeHandler.distribute(address(stableToken));
+  }
+
+  function test_CantDistributeWhenFrozen() public setUpBeneficiary activateToken {
+    freezer.freeze(address(feeHandler));
+    vm.expectRevert("can't call when contract is frozen");
+    feeHandler.distribute(address(stableToken));
+  }
+
+  function test_DoesntDistributeWhenBalanceIsZero()
+    public
+    setBurnFraction
+    setMaxSlippage
+    setUpBeneficiary
+    activateToken
+    fundFeeHandlerStable(1e18)
+  {
+    // If we uncomment this the test should fail
+    // feeHandler.sell(address(stableToken));
+    vm.recordLogs();
+    feeHandler.distribute(address(stableToken));
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+    assertEq(entries.length, 0);
+  }
+
+  function test_Distribute()
+    public
+    setBurnFraction
+    setMaxSlippage
+    setUpBeneficiary
+    activateToken
+    fundFeeHandlerStable(1e18)
+  {
+    feeHandler.sell(address(stableToken));
+
+    feeHandler.distribute(address(stableToken));
+
+    assertEq(stableToken.balanceOf(address(feeHandler)), 0);
+    assertEq(stableToken.balanceOf(EXAMPLE_BENEFICIARY_ADDRESS), 2e17);
+  }
+}
+
+contract FeeHandlerBurnCelo is FeeHandlerFoundry {
+  modifier activateToken() {
```

Comment:
missing the stabletoken activation

Reply:
We don't need to activate the stabletoken here because we aren't using it.